source/routing/tests/routing_api_test.cpp

```diff
@@ -3378,3 +3379,297 @@ BOOST_FIXTURE_TEST_CASE(should_filter_late_journeys_with_empty_list, Night_bus_f
         nr::filter_late_journeys(journeys, filter_params)
     );
 }
+
+/**
+ * @brief This test aims to highlight the timeframe_duration option on a journey request
+ *
+ * The data for testing :
+ *
+ * Vehicle journey with multiple stops times
+ * date : 2018 03 09
+ *
+ *                   sp1                   sp2
+ *                  x------------------------x
+ *               - 08:00:00               - 08:05:00
+ *               - 08:10:00               - 08:15:00
+ *               - 08:20:00               - 08:25:00
+ *               - 08:30:00               - 08:35:00
+ *                             ... x20
+ *               - 08:00:00               - 08:05:00 (24H after the first)
+ *
+ */
+BOOST_AUTO_TEST_CASE(journeys_with_time_frame_duration) {
+    // Create Builder
+    ed::builder b("20180309");
+
+    // Create Area
+    b.sa("stop_area:sa1")("stop_point:sa1:s1", 2.39592, 48.84848, false);
+    b.sa("stop_area:sa3")("stop_point:sa3:s1", 2.36381, 48.86650, false);
+
+    // Create multi VJ
+    auto dep_time = "08:00:00"_t;
+    auto arr_time = "08:05:00"_t;
+    for (int nb = 0; nb < 20; ++nb) {
+        b.vj("A", "1", "", false, "vjC_" + std::to_string(nb))
+                ("stop_point:sa1:s1", dep_time + nb * "00:10::00"_t)
+                ("stop_point:sa3:s1", arr_time + nb * "00:10::00"_t);
+    }
+    // Add a VJ 24H after to test the time frame duration max limit (24H)
+    b.vj("A", "1", "", false, "vjC_out_of_24_bound")
+                ("stop_point:sa1:s1", "08:00:00"_t)
+                ("stop_point:sa3:s1", "08:05:00"_t);
+
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+    b.data->build_proximity_list();
+    auto prod_date = boost::gregorian::date(2018, 3, 9);
+    auto prod_len = boost::gregorian::days(1);
+    b.data->meta->production_date = boost::gregorian::date_period(prod_date, prod_len);
+
+    // create Raptor
+    nr::RAPTOR raptor(*(b.data));
+
+    // from
+    navitia::type::EntryPoint origin(navitia::type::Type_e::StopPoint, "stop_point:sa1:s1");
+
+    // to
+    navitia::type::EntryPoint destination(navitia::type::Type_e::StopPoint, "stop_point:sa3:s1");
+
+    std::vector<std::string> forbidden;
+    ng::StreetNetwork sn_worker(*b.data->geo_ref);
+
+    auto * data_ptr = b.data.get();
+
+    //-----------------------------------
+    // Case 1 :
```

Comment:
It would be cool to add a test `min_nb_journey=2&timeframe_duration=40*60` also

Reply:
done test added https://github.com/CanalTP/navitia/pull/2442/files#diff-d7f228dace402afdab826f03fcda090cR3689